types/atom/index.d.ts

```diff
@@ -2799,20 +2799,7 @@ export interface Workspace {
      *  the existing item for that URI will be activated. If no URI is given, or
      *  no registered opener can open the URI, a new empty TextEditor will be created.
      */
-    open(uri: string, options?: WorkspaceOpenOptions): Promise<object>;
-    /**
-     *  Opens the given item in Atom asynchronously. If the item is already open,
-     *  the existing item will be activated. If no item is given, a new empty TextEditor
-     *  will be created.
-     */
-    open<T extends ViewModel = ViewModel>(item: T, options?: WorkspaceOpenOptions):
-        Promise<T>;
```

Comment:
I think we should not remove these. The second method is a general function that supports more than just TextEditor.

Reply:
But there isn't any signature for any more generality. I figured that the types should accurately reflect the docs since they are versioned

Reply:
This is the function for the reference: https://github.com/atom/atom/blob/f1b0e3b47b30ed9379821b341b3337696b8a2a12/src/workspace.js#L1077

Reply:
@lierdakil Where does the second function come from?

Reply:
@aminya, which one is the second one? The generic one? I'll be honest, I don't know, this isn't my code as far as I can tell. But the types here accurately reflect the behaviour of the code -- of that I'm certain because I use this in my projects a lot. And I'm pretty sure this thing with passing a view model to `open` was announced somewhere, either on Atom blog or somewhere in the flight manual. This is the case of "somebody forgot to update jsdoc" apparently.

Reply:
Oh yeah, I've found it. https://blog.atom.io/2017/05/16/atom-1-17.html